### PR TITLE
Student can enter the class room using code

### PR DESCRIPTION
**Pull request by MonitorIizard**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -41,11 +41,9 @@ services:
...
-        environment:
-            - NODE_ENV=production
         restart: always
         ports:
-            - 3000:3000
\ No newline at end of file
+            - 3000:3000
+        env_file:
+            - .env
+
```
Comment (tonkaew131) [inline docker-compose.yml:49]: Don't change production server docker-compose file without a reason, you gonna break everything

File: backend/schema/database-schema.sql [hunk 2]
```diff
@@ -62,4 +63,11 @@ CREATE TABLE IF NOT EXISTS teach (
 
     added_by TEXT REFERENCES auth_user(id),
     added_at TIMESTAMPTZ NOT NULL DEFAULT NOW()
+);
+
+CREATE TABLE IF NOT EXISTS study (
+    classroom_id    TEXT NOT NULL REFERENCES classroom(id) ON DELETE CASCADE,
+    user_id         TEXT NOT NULL REFERENCES auth_user(id) ON DELETE CASCADE,
+    section         INTEGER[] NOT NULL,
```
Comment (tonkaew131) [inline backend/schema/database-schema.sql:71]: Why is it an array? can users have multiple sections? But we also have a feature to create groups like in the class diagram.

File: backend/src/routes/classroom.ts [hunk 3]
```diff
@@ -90,4 +90,109 @@ export const classroomRoute = new Elysia({ prefix: '/classroom' })
...
+        const slug = slugRecords.slug;
+        return {
+            status: 'success',
+            message: 'You have joined the classroom.',
+            slug: slug,
+        };
+    })
+    .post(
+        '/create-invite-code',
+        async ({ body }: { body: { slug: string; section: number } }) => {
```
Comment (tonkaew131) [inline backend/src/routes/classroom.ts:168]: If you want to valid body request you should use elysia validator, but for type only its fine.

Here's the document about it, [Elysia Body Validation](https://elysiajs.com/validation/schema-type.html#body)

File: backend/src/routes/classroom.ts [hunk 4]
```diff
@@ -90,4 +90,109 @@ export const classroomRoute = new Elysia({ prefix: '/classroom' })
...
+    .post('/join-classroom', async ({ body, user, session, set }) => {
+        if (!user || !session) {
+            set.status = 401;
+            return {
+                status: 'error',
+                message: 'Unauthenticated, Please sign in and try again',
+            };
+        }
+
+        const joiningCode = (body as { classroomCode: string }).classroomCode;
```
Comment (tonkaew131) [inline backend/src/routes/classroom.ts:103]: For API body user snakecase for key

File: backend/src/routes/classroom.ts [hunk 5]
```diff
@@ -90,4 +90,109 @@ export const classroomRoute = new Elysia({ prefix: '/classroom' })
...
+            FROM study
+            WHERE classroom_id = ${classroomId} AND user_id = ${userId}
+        `;
+
+        if (isAlreadyJoined.length != 0) {
+            return {
+                status: 'error',
+                message: 'You have already joined the class.',
+            };
+        }
```
Comment (tonkaew131) [inline backend/src/routes/classroom.ts:145]: So user can join 1 section / 1 classroom? Why we have to have section as array?

File: learn-arai/components/context/SlugContext.tsx [hunk 6]
```diff
@@ -0,0 +1,5 @@
+import { createContext } from 'react';
+
+const SlugContext = createContext('');
```
Comment (tonkaew131) [inline learn-arai/components/context/SlugContext.tsx:3]: Great job on this

File: docker-compose.yml (hunk 1 above)
Comment (MonitorIizard) [reply docker-compose.yml:49]: fixed

File: backend/src/routes/classroom.ts (hunk 3 above)
Comment (MonitorIizard) [reply backend/src/routes/classroom.ts:168]: fixed

File: backend/src/routes/classroom.ts (hunk 4 above)
Comment (MonitorIizard) [reply backend/src/routes/classroom.ts:103]: fixed

File: backend/schema/database-schema.sql (hunk 2 above)
Comment (MonitorIizard) [reply backend/schema/database-schema.sql]: removed the section attribute

File: backend/src/routes/classroom.ts (hunk 5 above)
Comment (MonitorIizard) [reply backend/src/routes/classroom.ts:145]: removed section

**Comment by MonitorIizard** [comment]
11 March
-[ ] change create invitation code invite to `input chip`
- [ ] edit backend api.

**Comment by tonkaew131** [comment]
Going to implement Group system here

**Comment by tonkaew131** [comment]
#41 

- [ ] Including create invite code
- [ ] Manage group / user

**Comment by tonkaew131** [comment]
#41 Add Classroom' settings page

**Review by tonkaew131 — APPROVED** [review]

**Review by tonkaew131 — APPROVED** [review]
Nice

**Review by Gapraomoosap — APPROVED** [review]

**Pull request merged**